File: backend/schemas/lap.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LapCreate(BaseModel):
    session_id: uuid.UUID
    lap_number: int
    lap_time_ms: int
    sector1_ms: int | None = None
    sector2_ms: int | None = None
    sector3_ms: int | None = None
    max_speed_kmh: float | None = None


class LapResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    session_id: uuid.UUID
    lap_number: int
    lap_time_ms: int
    sector1_ms: int | None
    sector2_ms: int | None
    sector3_ms: int | None
    max_speed_kmh: float | None
    ai_summary: str | None
    ai_recommendations: list[str] | None
    created_at: datetime

File: frontend/src/types/Lap.ts
export interface Lap {
  id: string
  session_id: string
  lap_number: number
  lap_time_ms: number
  sector1_ms: number | null
  sector2_ms: number | null
  sector3_ms: number | null
  max_speed_kmh: number | null
  ai_summary: string | null
  ai_recommendations: string[] | null
  created_at: string
}
